### PR TITLE
Modernize/refresh plugin

### PR DESCRIPTION
**Pull request by jonesbusy**
Hi,

This is an attempt to refresh and modernize the plugin configuration with recent core version, plugin bom, etc...

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

Regards,

### CODE REVIEW

**Comment by jonesbusy** [comment]
Builds are failing due to change of Jenkinsfile

```
11:15:52  ‘Jenkinsfile’ has been modified in an untrusted revision
```

**Comment by segaura** [comment]
Looking at the problematic step logs, it shows 

`Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.jenkins-ci.main:jenkins-bom:pom:2.107.2 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/)`

and it is looking for a way too old version which is present no more. Maybe moving to a `2.364` or `2.373` version could solve this. If this is not an useful hint, forget about it, I am here starting to study this plugin and simply noticed this issue :-)

**Comment by jonesbusy** [comment]
> Looking at the problematic step logs, it shows 
> 
> `Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.jenkins-ci.main:jenkins-bom:pom:2.107.2 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/)`
> 
> and it is looking for a way too old version which is present no more. Maybe moving to a `2.364` or `2.373` version could solve this. If this is not an useful hint, forget about it, I am here starting to study this plugin and simply noticed this issue :-)

I've updated the Jenkinsfile. But I'm no maintainer of this plugin that's why it still  use the default one (Cannot modify Jenkinsfile from untrusted source.

We need a maintainer here to perform those changes.

**Comment by jonesbusy** [comment]
Friendly reminder @jenkinsci/gatling-plugin-developers :)

File: pom.xml [hunk 1]
```diff
@@ -69,77 +67,53 @@
...
-			<version>1.14</version>
 		</dependency>
 		<dependency>
 			<groupId>org.jenkins-ci.plugins.workflow</groupId>
 			<artifactId>workflow-step-api</artifactId>
-			<version>1.14</version>
 		</dependency>
 		<!-- Required due to JENKINS-51013 in plugin compat tester -->
-		<dependency>
+		<!-- <dependency>
```
Comment (slandelle) [inline pom.xml:101]: Does this mean that this dependency is no longer required? If so, please remove the now commented blocks.

**Review by slandelle — CHANGES_REQUESTED** [review]
Thanks!
Just one small nit.
Also, I've pushed the modified Jenkinsfile. Could you please remove it from your PR?

File: pom.xml (hunk 1 above)
Comment (jonesbusy) [reply pom.xml]: Seems the mentioned issue was fixed and doesn't appear using more recent parent. There is also no reason to have a hard dependency on this plugin

**Comment by jonesbusy** [comment]
Thanks! I've rebased the PR and fix comments

File: pom.xml (hunk 1 above)
Comment (slandelle) [reply pom.xml]: thx

**Comment by slandelle** [comment]
Merged. Thanks a lot!

**Comment by jonesbusy** [comment]
Hi!

@slandelle Can we cut a release including this change ? (Or even better include continuous delivery of Jenkins plugin : https://www.jenkins.io/doc/developer/publishing/releasing-cd/)

Thanks!

**Comment by slandelle** [comment]
@jonesbusy My understanding was that this change was only an internal/technical one, without any user impact. Am I missing something?

**Comment by jonesbusy** [comment]
Yes only bumping core version to avoid implicit plugin dependencies

**Comment by slandelle** [comment]
Release failure: `authentication failed for https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/gatling/1.3.1/gatling-1.3.1.pom, status: 401 Unauthorized -> [Help 1]`.

It was working before, but I haven't release for a long time. Maybe Jenkins has broken my creds, sadly no time to investigate atm.

**Comment by vittoriocanilli** [comment]
@slandelle any chance to have a new release soon? The last one is over 4 years old and it causes some implicit plugin dependencies. Thanks in advance!

**Comment by slandelle** [comment]
IIRC, I tried to release last year but it failed as the release process had completely changed. I'll be off for the next 2 weeks, I'll try to have a look when I'm back.

**Comment by vittoriocanilli** [comment]
@slandelle I hope you had great holidays: any chance to have a new release soon? Thanks in advance!

**Comment by slandelle** [comment]
Hello @vittoriocanilli 
I tried to release some time ago, but I realized that the release process here had completely changed.
I have no clue about how things are supposed to work nowadays.

Then, we at Gatling won't be maintaining this plugin ourselves anymore.
So I would be fine with handing it over to other maintainers. 
If someone is interested, please chime in.

**Comment by jonesbusy** [comment]
Proess is describe here https://www.jenkins.io/doc/developer/publishing/releasing-cd/#setting-up-automated-plugin-release

Can you at least mark the plugin for adoption : https://www.jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/#how-can-i-mark-a-plugin-for-adoption

Thanks

**Comment by vittoriocanilli** [comment]
@jonesbusy It looks like the plugin is now up for adoption 🥳

**Pull request merged**